paymentsheet/src/main/java/com/stripe/android/customersheet/CustomerAdapter.kt

```diff
@@ -150,26 +169,63 @@ interface CustomerAdapter {
             }
         }
     }
+
+    @ExperimentalCustomerSheetApi
+    @RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
+    @JvmInline
+    value class Result<out T> internal constructor(
```

Comment:
I'm curious to see what this looks like as far as public API (in our .api file) when we make this public. value class is a little bit scary in terms of how do we maintain compatibility. But kotlin does it, so it must not be too hard. It's also a little sad we can't have access to all the kotlin.Result methods/extension methods.

Nothing actionable here, just me thinking out loud.